### PR TITLE
Fix #10659: [BUG] DefaultAzureCredential cluttering log with errors

### LINKED ISSUE #10659: [BUG] DefaultAzureCredential cluttering log with errors
**Describe the bug**
I'm using this to access a blob container
`_blobContainerClient = new BlobContainerClient(new Uri(containerUrl), new DefaultAzureCredential());`

This is running in a container on AKS with Pod Identity - so ManagedIdentityCredential should be the one that is used. 

However, since this is only [second in the list,](https://docs.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredential?view=azure-dotnet) the attempt to use EnvironmentCredential obviously fails. So far, so good. The issue is that it is not failing silently to the next one but instead cluttering my log (which I'm sending to Application Insights):

image

(disregard the 404 error for the blob. I'm checking if the blob does exist or not, so the 404 there is expected)

**Expected behavior**
When using DefaultAzureCredential I would expect that all options are silently being tried in their order. Only if all fail, I'd expect an exception to be thrown.

**Actual behavior (include Exception or Stack Trace)**
Error: 

> Azure.Identity.CredentialUnavailableException: EnvironmentCredential authentication unavailable. Environment variables are not fully configured. 

**To Reproduce**
Steps to reproduce the behavior (include a code snippet, screenshot, or any additional information that might help us reproduce the issue)

1. Use DefaultAzureCredential() with a method that is not the first one in the list (EnvironmentCredential )

**Environment:**
 - Name and version of the Library package used: Azure.Identity 1.2.0-preview.1
 - Hosting platform or OS and .NET runtime version: .NET Core 3.1 in Linux Container on AKS
 - IDE and version : VS 16.4.6

Also a general question: If I know that I will be using Managed Identity, is it best practice to use that specifically instead of the DefaultCredential?

### PR DESCRIPTION
**Pull request by AlexanderSher**

### CODE REVIEW

File: sdk/identity/Azure.Identity/src/DateTimeHelpers.netfx.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            Thread.Sleep(1);
+            Interlocked.Exchange(ref _timeSync, new TimeSync());
+        };
+
+        /// <summary>
+        /// Returns high resolution (1 DateTime tick) current UTC DateTime.
+        /// </summary>
+        private static DateTime GetUtcNowImplementation()
+        {
+            // DateTime.UtcNow accuracy on .NET Framework is ~16ms, this method
```
Comment (pakrym) [inline sdk/identity/Azure.Identity/src/DateTimeHelpers.netfx.cs:33]: Do we care about 16ms error?

File: sdk/identity/Azure.Identity/src/CredentialPipeline.cs [hunk 2]
```diff
@@ -2,29 +2,38 @@
...
 using Microsoft.Identity.Client;
 
 namespace Azure.Identity
 {
    internal class CredentialPipeline
     {
-        private static readonly Lazy<CredentialPipeline> s_Singleton = new Lazy<CredentialPipeline>(() => new CredentialPipeline(new TokenCredentialOptions()));
+        private static readonly Lazy<CredentialPipeline> s_singleton = new Lazy<CredentialPipeline>(() => new CredentialPipeline(new TokenCredentialOptions()));
+
+        private static readonly AsyncLocal<IScopeHandler> s_scopeHandler = new AsyncLocal<IScopeHandler>();
```
Comment (pakrym) [inline sdk/identity/Azure.Identity/src/CredentialPipeline.cs:19]: Can we avoid the AsyncLocal?

File: sdk/identity/Azure.Identity/src/DateTimeHelpers.netfx.cs (hunk 1 above)
Comment (AlexanderSher) [reply sdk/identity/Azure.Identity/src/DateTimeHelpers.netfx.cs:33]: `System.Diagnostics.Activity` will have wrong `Duration` if we don't use the same precision they use:
https://github.com/dotnet/runtime/blob/21ea59f90044f4f662117fc43022f07eafb485cd/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L521

Not sure how critical is that, but IMO it is good to have consistent behavior of `UtcNow` across different frameworks.

File: sdk/identity/Azure.Identity/src/CredentialPipeline.cs (hunk 2 above)
Comment (AlexanderSher) [reply sdk/identity/Azure.Identity/src/CredentialPipeline.cs:19]: All credentials share the same pipeline which is owned by factory. Let me try to hold the handler in `CredentialPipeline` itself.

Comment (AlexanderSher) [reply sdk/identity/Azure.Identity/src/CredentialPipeline.cs:19]: It works, but there is a problem with concurrent calls. I can address it in a fix for https://github.com/Azure/azure-sdk-for-net/issues/13044

File: sdk/identity/Azure.Identity/src/DateTimeHelpers.netfx.cs (hunk 1 above)
Comment (schaabs) [reply sdk/identity/Azure.Identity/src/DateTimeHelpers.netfx.cs:33]: Is this done in `Azure.Core` already? If not I agree that it's adds a good bit of complexity for something which doesn't really impact users.

File: sdk/identity/Azure.Identity/api/Azure.Identity.netstandard2.0.cs [hunk 3]
```diff
@@ -57,9 +57,8 @@ public partial class CredentialUnavailableException : Azure.Identity.Authenticat
     }
     public partial class DefaultAzureCredential : Azure.Core.TokenCredential
     {
-        public DefaultAzureCredential() { }
         public DefaultAzureCredential(Azure.Identity.DefaultAzureCredentialOptions options) { }
-        public DefaultAzureCredential(bool includeInteractiveCredentials) { }
+        public DefaultAzureCredential(bool includeInteractiveCredentials = false) { }
```
Comment (schaabs) [inline sdk/identity/Azure.Identity/api/Azure.Identity.netstandard2.0.cs:61]: I believe you were planning to revert this change no?

Comment (AlexanderSher) [reply sdk/identity/Azure.Identity/api/Azure.Identity.netstandard2.0.cs:61]: This is reverting the change, the change itself was part of another PR that has been merged already.

File: sdk/identity/Azure.Identity/src/DateTimeHelpers.netfx.cs (hunk 1 above)
Comment (AlexanderSher) [reply sdk/identity/Azure.Identity/src/DateTimeHelpers.netfx.cs:33]: Ok, reverting it.

Comment (AlexanderSher) [reply sdk/identity/Azure.Identity/src/DateTimeHelpers.netfx.cs]: Reverted

File: sdk/identity/Azure.Identity/api/Azure.Identity.netstandard2.0.cs (hunk 3 above)
Comment (schaabs) [reply sdk/identity/Azure.Identity/api/Azure.Identity.netstandard2.0.cs:61]: sorry, I read it backwards :smile:.

**Review by schaabs — APPROVED** [review]
